diff -up shadow-4.15.0/src/login.c.3~ shadow-4.15.0/src/login.c
--- shadow-4.15.0/src/login.c.3~	2024-03-08 22:27:04.000000000 +0100
+++ shadow-4.15.0/src/login.c	2024-03-09 02:20:17.568202890 +0100
@@ -24,6 +24,7 @@
 #include <sys/stat.h>
 #include <sys/ioctl.h>
 #include <assert.h>
+#include <malloc.h>
 
 #include "alloc.h"
 #include "attr.h"
@@ -739,6 +740,7 @@ int main (int argc, char **argv)
 			}
 #endif
 
+                        malloc_trim(0);
 			retcode = pam_authenticate (pamh, 0);
 
 			get_pam_user (&pam_user);
