diff -up shadow-4.14.0/src/login.c.4~ shadow-4.14.0/src/login.c
--- shadow-4.14.0/src/login.c.4~	2023-08-10 16:30:00.000000000 +0200
+++ shadow-4.14.0/src/login.c	2023-08-16 14:47:21.961716040 +0200
@@ -24,6 +24,7 @@
 #include <sys/stat.h>
 #include <sys/ioctl.h>
 #include <assert.h>
+#include <malloc.h>
 
 #include "alloc.h"
 #include "defines.h"
@@ -744,6 +745,7 @@ int main (int argc, char **argv)
 			}
 #endif
 
+                        malloc_trim(0);
 			retcode = pam_authenticate (pamh, 0);
 
 			get_pam_user (&pam_user);
