diff -up shadow-4.10/libmisc/Makefile.am.6~ shadow-4.10/libmisc/Makefile.am
--- shadow-4.10/libmisc/Makefile.am.6~	2021-07-25 19:21:42.000000000 +0200
+++ shadow-4.10/libmisc/Makefile.am	2021-12-21 16:43:53.419165634 +0100
@@ -77,3 +77,4 @@ if WITH_BTRFS
 libmisc_la_SOURCES += btrfs.c
 endif
 
+libmisc_la_LIBADD = -lpam
diff -up shadow-4.10/libsubid/Makefile.am.6~ shadow-4.10/libsubid/Makefile.am
--- shadow-4.10/libsubid/Makefile.am.6~	2021-12-21 16:43:53.419165634 +0100
+++ shadow-4.10/libsubid/Makefile.am	2021-12-21 16:44:25.906909913 +0100
@@ -26,7 +26,7 @@ MISCLIBS = \
 libsubid_la_LIBADD = \
 	$(top_builddir)/lib/libshadow.la \
 	$(top_builddir)/libmisc/libmisc.la \
-	$(MISCLIBS) -ldl
+	$(MISCLIBS) -ldl -lpam -lpam_misc
 
 AM_CPPFLAGS = \
 	-I${top_srcdir}/lib \
