diff -Naur shadow-4.8/src/useradd.c shadow-4.8.tpg/src/useradd.c
--- shadow-4.8/src/useradd.c	2019-12-01 16:52:32.000000000 +0000
+++ shadow-4.8.tpg/src/useradd.c	2019-12-03 12:52:04.626034152 +0000
@@ -331,7 +331,7 @@
 	              user_name, AUDIT_NO_ID,
 	              SHADOW_AUDIT_FAILURE);
 #endif
-	SYSLOG ((LOG_INFO, "failed adding user '%s', data deleted", user_name));
+	SYSLOG ((LOG_INFO, "failed adding user '%s', exit code: %d", user_name, code));
 	exit (code);
 }
 
