diff -up shadow-4.14.0/libmisc/Makefile.am.3~ shadow-4.14.0/libmisc/Makefile.am
--- shadow-4.14.0/libmisc/Makefile.am.3~	2023-08-16 13:55:40.443139111 +0200
+++ shadow-4.14.0/libmisc/Makefile.am	2023-08-16 13:57:14.840060090 +0200
@@ -112,3 +112,5 @@ libmisc_la_SOURCES += \
 	readpassphrase.h \
 	readpassphrase.c
 endif
+
+libmisc_la_LIBADD = -lpam
diff -up shadow-4.14.0/libsubid/Makefile.am.3~ shadow-4.14.0/libsubid/Makefile.am
--- shadow-4.14.0/libsubid/Makefile.am.3~	2023-03-13 21:58:56.000000000 +0100
+++ shadow-4.14.0/libsubid/Makefile.am	2023-08-16 13:55:40.443139111 +0200
@@ -22,7 +22,7 @@ MISCLIBS = \
 libsubid_la_LIBADD = \
 	$(top_builddir)/lib/libshadow.la \
 	$(top_builddir)/libmisc/libmisc.la \
-	$(MISCLIBS) -ldl
+	$(MISCLIBS) -ldl -lpam -lpam_misc
 
 AM_CPPFLAGS = \
 	-I${top_srcdir}/lib \
