diff -up shadow-4.12/libmisc/copydir.c.3~ shadow-4.12/libmisc/copydir.c
--- shadow-4.12/libmisc/copydir.c.3~	2022-08-11 16:34:20.000000000 +0200
+++ shadow-4.12/libmisc/copydir.c	2022-08-11 20:48:42.906183545 +0200
@@ -216,6 +216,7 @@ int copy_tree (const char *src_root, con
 	bool set_orig = false;
 	struct dirent *ent;
 	DIR *dir;
+	int len = 0;
 
 	if (copy_root) {
 		struct stat sb;
@@ -276,6 +277,18 @@ int copy_tree (const char *src_root, con
 			char *dst_name;
 			size_t src_len = strlen (ent->d_name) + 2;
 			size_t dst_len = strlen (ent->d_name) + 2;
+
+			/*
+			 * Skip the ".rpmsave", ".rpmnew" and ".rpmorig" entries
+			 */
+			
+			len = strlen (ent->d_name);
+			if (len > 8 && (strcmp ((ent->d_name+len-8) , ".rpmsave") == 0 ||
+				strcmp ((ent->d_name+len-8), ".rpmorig") == 0))
+				continue;
+			if (len > 7 && strcmp ((ent->d_name+len-7), ".rpmnew") == 0)
+			    continue;
+
 			src_len += strlen (src_root);
 			dst_len += strlen (dst_root);
 
