diff -up shadow-4.14.1/lib/Makefile.am.3~ shadow-4.14.1/lib/Makefile.am
--- shadow-4.14.1/lib/Makefile.am.3~	2023-10-05 23:38:24.259303507 +0200
+++ shadow-4.14.1/lib/Makefile.am	2023-10-05 23:39:29.748091998 +0200
@@ -7,6 +7,7 @@ noinst_LTLIBRARIES = libshadow.la
 
 if USE_PAM
 LIBCRYPT_PAM = $(LIBCRYPT)
+libshadow_la_LIBADD = -lpam
 else
 LIBCRYPT_PAM =
 endif
diff -up shadow-4.14.1/libsubid/Makefile.am.3~ shadow-4.14.1/libsubid/Makefile.am
--- shadow-4.14.1/libsubid/Makefile.am.3~	2023-10-05 23:38:24.260303519 +0200
+++ shadow-4.14.1/libsubid/Makefile.am	2023-10-05 23:39:51.481350293 +0200
@@ -21,7 +21,7 @@ MISCLIBS = \
 
 libsubid_la_LIBADD = \
 	$(top_builddir)/lib/libshadow.la \
-	$(MISCLIBS) -ldl
+	$(MISCLIBS) -ldl -lpam -lpam_misc
 
 AM_CPPFLAGS = \
 	-I${top_srcdir}/lib \
