diff -up shadow-4.12.3/libmisc/copydir.c.3~ shadow-4.12.3/libmisc/copydir.c
--- shadow-4.12.3/libmisc/copydir.c.3~	2022-08-23 02:13:57.256609249 +0200
+++ shadow-4.12.3/libmisc/copydir.c	2022-08-23 02:17:46.525669846 +0200
@@ -321,6 +321,18 @@ static int copy_tree_impl (const struct
 			char *dst_name;
 			size_t src_len = strlen (ent->d_name) + 2;
 			size_t dst_len = strlen (ent->d_name) + 2;
+
+			/*
+			 * Skip the ".rpmsave", ".rpmnew" and ".rpmorig" entries
+			 */
+			
+			const int len = strlen (ent->d_name);
+			if (len > 8 && (strcmp ((ent->d_name+len-8) , ".rpmsave") == 0 ||
+				strcmp ((ent->d_name+len-8), ".rpmorig") == 0))
+				continue;
+			else if (len > 7 && strcmp ((ent->d_name+len-7), ".rpmnew") == 0)
+				continue;
+
 			src_len += strlen (src->full_path);
 			dst_len += strlen (dst->full_path);
 
