--- shadow-4.1.5.1/src/Makefile.am.orig	2012-07-12 13:32:21.500031900 +0100
+++ shadow-4.1.5.1/src/Makefile.am	2012-07-12 13:38:35.633678734 +0100
@@ -55,14 +55,14 @@
 endif
 
 if WITH_TCB
-suidubins -= passwd
-shadowsgidubins = passwd
+	suidubins -= passwd
+	shadowsgidubins = passwd
 endif
 
 LDADD          = $(INTLLIBS) \
-		 $(LIBTCB) \
 		 $(top_builddir)/libmisc/libmisc.a \
-		 $(top_builddir)/lib/libshadow.la
+		 $(top_builddir)/lib/libshadow.la \
+		 $(LIBTCB)
 AM_CPPFLAGS    = -DLOCALEDIR=\"$(datadir)/locale\"
 
 if ACCT_TOOLS_SETUID
--- shadow-4.1.5.1/src/userdel.c.orig	2012-07-12 13:39:56.881283335 +0100
+++ shadow-4.1.5.1/src/userdel.c	2012-07-12 13:40:38.020574816 +0100
@@ -798,7 +798,7 @@
 {
 	char *buf;
 	int ret = 0;
-	size_t bufsize = (sizeof TCB_DIR) + strlen (user_name) + 2;
+	size_t buflen = (sizeof TCB_DIR) + strlen (user_name) + 2;
 
 	if (!getdef_bool ("USE_TCB")) {
 		return 0;
